### PR TITLE
python3Packages.tesserocr: init at 2.5.0

### PR DESCRIPTION
**Pull request by schonfinkel**
###### Motivation for this change

This is an attempt of adding [tesserocr](https://pypi.org/project/tesserocr/) to nixpkgs, there are already some wrappers for tesseract in `nixpkgs`, but I need this one and it seems there's no derivation yet (so I put myself as maintainer).

Any comments on this derivation are always welcome. I'm currently not sure on how to test a python library, but the derivation compiles.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -4666,6 +4666,12 @@
     githubId = 2881922;
     name = "Francis St-Amour";
   };
+  mtrsk = {
+    email = "marcos.schonfinkel@protonmail.com";
+    github = "mtrsk";
+    githubId = 16356569;
+    name = "Marcos Benevides";
+  };
```
Comment (jonringer) [inline maintainers/maintainer-list.nix]: please split this out into a different commit:
```
maintainers: add mtrsk
```

File: pkgs/development/python-modules/tesserocr/default.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+buildPythonPackage rec {
+  pname = "tesserocr";
+  version = "2.5.0";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0nj33mwvppacy6p5mqk9a4x26hx9ailshgad84ks60wyms6rgjiv";
+  };
+
+  buildInputs = [ cython leptonica tesseract pkg-config ];
```
Comment (jonringer) [inline pkgs/development/python-modules/tesserocr/default.nix]:
```suggestion
  nativeBuildInputs = [ cython leptonica tesseract pkg-config ];
```

File: pkgs/development/python-modules/tesserocr/default.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  propagatedBuildInputs = [ pillow ];
+
+  # the package doesn't have any tests.
+  doCheck = false;
+
+  meta = with lib; {
+    homepage = https://pypi.org/project/tesserocr/;
+    license = licenses.mit;
+    description = "A simple, Pillow-friendly, wrapper around the tesseract-ocr API for Optical Character Recognition (OCR)";
+    maintainers = with maintainers; [ mtrsk ];
```
Comment (jonringer) [inline pkgs/development/python-modules/tesserocr/default.nix]: kinda minor, but most expressions go
```
description
homepage
license
maintainers
```

not super import though

File: pkgs/development/python-modules/tesserocr/default.nix (hunk 2 above)
Comment (mtrsk) [reply pkgs/development/python-modules/tesserocr/default.nix]: Putting `tesseract` and `leptonica` in `nativeBuildInputs` makes the build fail.

Comment (jonringer) [reply pkgs/development/python-modules/tesserocr/default.nix]: sounds good, i was only sure about pkg-config and cython :)

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (mtrsk) [reply maintainers/maintainer-list.nix]: done! :)

File: pkgs/development/python-modules/tesserocr/default.nix [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  pname = "tesserocr";
+  version = "2.5.0";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0nj33mwvppacy6p5mqk9a4x26hx9ailshgad84ks60wyms6rgjiv";
+  };
+
+  buildInputs = [ leptonica tesseract ];
+  nativeBuildInputs = [ cython pkg-config ];
```
Comment (jonringer) [inline pkgs/development/python-modules/tesserocr/default.nix]: do you mind switching the order of build and nativeBuild inputs. 

this is a pretty typical ordering to a file:
https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/python-modules/pandas/default.nix

File: pkgs/development/python-modules/tesserocr/default.nix [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+  propagatedBuildInputs = [ pillow ];
+
+  # the package doesn't have any tests.
+  doCheck = false;
+
+  meta = with lib; {
+    description = "A simple, Pillow-friendly, wrapper around the tesseract-ocr API for Optical Character Recognition (OCR)";
+    homepage = https://pypi.org/project/tesserocr/;
```
Comment (jonringer) [inline pkgs/development/python-modules/tesserocr/default.nix]: Thats just their page on pypi
```suggestion
    homepage = "https://github.com/sirfz/tesserocr";
```

File: pkgs/development/python-modules/tesserocr/default.nix [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+    inherit pname version;
+    sha256 = "0nj33mwvppacy6p5mqk9a4x26hx9ailshgad84ks60wyms6rgjiv";
+  };
+
+  buildInputs = [ leptonica tesseract ];
+  nativeBuildInputs = [ cython pkg-config ];
+  propagatedBuildInputs = [ pillow ];
+
+  # the package doesn't have any tests.
+  doCheck = false;
```
Comment (jonringer) [inline pkgs/development/python-modules/tesserocr/default.nix]: do you mind checking out from GitHub so we can pull down tests?
```suggestion
  doCheck = false;
```

tests are really important to make sure you don't get broken by bumps in upstream dependencies

Comment (mtrsk) [reply pkgs/development/python-modules/tesserocr/default.nix]: Removed the `doCheck`, since the docs say it's `true` by default, package builds fine!

Comment (jonringer) [reply pkgs/development/python-modules/tesserocr/default.nix]: for buildPythonPackage, this is true

**Comment by jonringer** [comment]
the commit message should reflect the install path (what you put into nix-build), so your commit should be:

```
python3Packages.tesserocr: init at 2.5.0
```

**Review by jonringer — APPROVED** [review]
diff LGTM
commits LGTM

```
https://github.com/NixOS/nixpkgs/pull/74389
3 package were built:
python27Packages.tesserocr python37Packages.tesserocr python38Packages.tesserocr
```

**Comment by jonringer** [comment]
@GrahamcOfBorg build python27Packages.tesserocr python37Packages.tesserocr python38Packages.tesserocr

**Comment by jonringer** [comment]
can you mark this as linux only? unless you want to trouble shoot why darwin is failing on dlopen

**Comment by jonringer** [comment]
@GrahamcOfBorg build python27Packages.tesserocr python37Packages.tesserocr python38Packages.tesserocr

**Comment by jonringer** [comment]
LGTM
